src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1960,14 +1960,43 @@ fn primitive_impl(cx: &DocContext<'_>, path_str: &str) -> Option<&'static SmallV
 fn strip_generics_from_path(path_str: &str) -> Result<String, ResolutionFailure<'static>> {
     let mut stripped_segments = vec![];
 
-    for segment in path_str.split("::") {
-        let stripped_segment = strip_generics_from_path_segment(segment)?;
+    let path: Vec<char> = path_str.chars().collect();
```

Comment:
Does this need to copy the string again? I think you could iterate over `path_str.char_indices().peekable()`.

Reply:
What do you mean by "again"?

Reply:
Well, you haven't copied it yet here, but rustdoc copies it in lots of other places 😅